packages/python/plotly/plotly/express/_imshow.py

```diff
@@ -419,7 +519,23 @@ def imshow(
         layout_patch["title_text"] = args["title"]
     elif args["template"].layout.margin.t is None:
         layout_patch["margin"] = {"t": 60}
-    fig = go.Figure(data=trace, layout=layout)
+
+    frame_list = []
+    for index, trace in enumerate(traces):
+        if (facet_col and index < nrows * ncols) or index == 0:
+            fig.add_trace(trace, row=nrows - index // ncols, col=index % ncols + 1)
+    if animation_frame is not None:
+        for i, index in zip(range(nslices_animation), animation_slices):
+            frame_list.append(
+                dict(
+                    data=traces[nslices_facet * i : nslices_facet * (i + 1)],
+                    layout=layout,
+                    name=str(index),
+                )
+            )
+    if animation_frame:
+        fig.frames = frame_list
+    fig.update_layout(layout)
```

Comment:
it's a bit odd to have `layout` and `layout_patch` here... it was odd before I guess but might be worth looking at merging them together earlier?